synapse/storage/database.py

```diff
@@ -1355,7 +1362,7 @@ def simple_upsert_many_txn_emulated(
             _keys = {x: y for x, y in zip(key_names, keyv)}
             _vals = {x: y for x, y in zip(value_names, valv)}
 
-            self.simple_upsert_txn_emulated(txn, table, _keys, _vals)
+            self.simple_upsert_txn_emulated(txn, table, _keys, _vals, lock=lock)
```

Comment:
I'm guessing this works fine since we lock the table (as part of the transaction) in the first iterate of the loop, but then it stays locked until the end of the transaction?

1. Is this accurate?
2. If so, is there overhead to locking it for every value? (Should we only lock the table on the first iteration of the loop?)

Reply:
@reivilibre judging from your commits my comment above was accurate, I think?